### PR TITLE
Add basic documentation on Kubernetes Service Binding

### LINKED ISSUE #15626: Missing documentation about k8s Service Bindings
There is some support for k8s Service Bindings (https://github.com/k8s-service-bindings/spec) in a few (most?) of our jdbc data sources, but it's not mentioned in https://quarkus.io/guides/deploying-to-kubernetes and I'd like to see a section that explains that it's supported and how a minimal file system tree config will map to, say, a postgres quarkus configuration.

We should also list the extensions that support it ATM, because even if it will become outdated and not list newer extensions, at least it tells me which are expected to work.

Probably should also list the config keys that are loaded?

CC @geoand

### PR DESCRIPTION
**Pull request by geoand**
Also changes the default value of `quarkus.kubernetes-service-binding.enabled` to `true`

Resolves: #15626

### CODE REVIEW

File: docs/src/main/asciidoc/deploying-to-kubernetes.adoc [hunk 1]
```diff
@@ -1185,3 +1185,69 @@ The provided replicas <1>, labels <2> and environment variables <3>  were retain
...
+Currently, the following Quarkus extensions support this feature:
+
+* quarkus-jdbc-mariadb
+* quarkus-jdbc-mssql
+* quarkus-jdbc-mysql
+* quarkus-jdbc-postgresql
+
+This list of extensions will grow as more services with supported bindings become available on Kubernetes.
+
+To enable Service Binding support, in addition to one of currently supported extensions, the `quarkus-kubernetes-service-binding` extension needs to be added to the application dependencies as well as
```
Comment (gsmet) [inline docs/src/main/asciidoc/deploying-to-kubernetes.adoc:1204]:
```suggestion
To enable Service Binding support, in addition to one of the currently supported extensions, the `quarkus-kubernetes-service-binding` extension needs to be added to the application dependencies as well as
```

**Review by gsmet — DISMISSED** [review]
Suggested a small improvement, otherwise it looks very good!

File: docs/src/main/asciidoc/deploying-to-kubernetes.adoc [hunk 2]
```diff
@@ -1185,3 +1185,69 @@ The provided replicas <1>, labels <2> and environment variables <3>  were retain
...
+having the `quarkus.kubernetes-service-binding.enabled` configuration property set to `true`.
```
Comment (FroMage) [inline docs/src/main/asciidoc/deploying-to-kubernetes.adoc:1205]: While this isn't about documentation, I would not have suspected this without docs.

Is there a reason why this isn't on by default when the extension is used?

File: docs/src/main/asciidoc/deploying-to-kubernetes.adoc [hunk 3]
```diff
@@ -1185,3 +1185,69 @@ The provided replicas <1>, labels <2> and environment variables <3>  were retain
...
+----
+quarkus.kubernetes-service-binding.enabled=true
+quarkus.hibernate-orm.database.generation=validate
+----
+
+You'll notice that all the configuration for the datasource is absent, as it will be auto-discovered by Quarkus when the application is deployed to Kubernetes.
+
+=== How does it work?
+
+When the application is deployed to an appropriately configured Kubernetes cluster, Kubernetes will create on the filesystem a set of files containing the runtime configuration (for example a file containing URL of database, another for the username).
```
Comment (FroMage) [inline docs/src/main/asciidoc/deploying-to-kubernetes.adoc:1241]:
```suggestion
When the application is deployed to an appropriately configured Kubernetes cluster, Kubernetes will create on the filesystem a set of files containing the runtime configuration (for example a file containing the URL of the database, another for the username).
```

File: docs/src/main/asciidoc/deploying-to-kubernetes.adoc (hunk 2 above)
Comment (geoand) [reply docs/src/main/asciidoc/deploying-to-kubernetes.adoc:1205]: To be honest, I didn't remember this either and on second look it did surprise me too.
Should I change the default to true?

Comment (FroMage) [reply docs/src/main/asciidoc/deploying-to-kubernetes.adoc:1205]: Yeah, after all other config sources have higher priority, so… yeah.

Comment (geoand) [reply docs/src/main/asciidoc/deploying-to-kubernetes.adoc]: Fixed

**Review by FroMage — APPROVED** [review]

**Pull request merged**